### PR TITLE
fix(storage): fix bloom filter's bits_per_key

### PR DESCRIPTION
**Pull request by KeXiangWang**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

1. fix bug of bloom filter. Now both the number of functions and the size of bloom filter is smaller than expected.
2. change default bloom_false_positive from 0.01 to 0.001.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

**Comment by KeXiangWang** [comment]
Here's the formula of calculate the size and k of a bloom filter:
image
m is the size of bloom filter, n is the number of the keys to be inserted, k is the number of the hash functions, p is the expected false positive rate.

Now, in our implementation, both k and m multiply an extra ln2(approximately 0.69). So our bloom filter's false positive rate sucks.

**Comment by KeXiangWang** [comment]
Here's the comparsion between before and after.
image
As we can see, after fix, the false positive rate is much lower.

And the default expected false positive rate is 0.01 now. If a key is in ranges of 10 sst. Then the false positive rate will be close to 10%. And this will severely affect our latency and throughput. So 0.001 would be much better, although the size of bloom filter will double. It worths it.

**Review by hzxa21 — APPROVED** [review]
LGTM. Thanks for the PR.

This is very subtle. Good catch!

File: docker/risingwave.toml [hunk 1]
```diff
@@ -14,7 +14,7 @@ checkpoint_frequency = 10
 shared_buffer_capacity_mb = 4096
```
Comment (fuyufjh) [inline docker/risingwave.toml:14]: (off-topic) I think we need to delete this file `docker/risingwave.toml` like #6814 did.

File: src/storage/src/hummock/sstable/bloom.rs [hunk 2]
```diff
@@ -156,4 +158,44 @@ mod tests {
...
+    fn false_positive_rate_case(
+        preset_key_count: usize,
+        test_key_count: usize,
+        expected_false_positive_rate: f64,
+    ) {
+        let mut key_list = vec![];
+
+        for i in 0..preset_key_count {
+            let k = Bytes::from(format!("{:032}", i));
+            let h = farmhash::fingerprint32(&k);
```
Comment (Li0k) [inline src/storage/src/hummock/sstable/bloom.rs:171]: can we unify the hash function to `xxh` ? like sstable::builder

File: src/storage/src/hummock/sstable/bloom.rs [hunk 3]
```diff
@@ -156,4 +158,44 @@ mod tests {
...
+        }
+
+        let bits_per_key = Bloom::bloom_bits_per_key(key_list.len(), expected_false_positive_rate);
+        let vec = Bloom::build_from_key_hashes(&key_list, bits_per_key);
+        let filter = Bloom::new(&vec);
+
+        let mut true_count = 0;
+        for i in preset_key_count..preset_key_count + test_key_count {
+            let k = Bytes::from(format!("{:032}", i));
+            let h = farmhash::fingerprint32(&k);
```
Comment (Li0k) [inline src/storage/src/hummock/sstable/bloom.rs:182]: ditto

**Review by Li0k — APPROVED** [review]
Rest LGTM. Thanks for the PR.

**Review by Little-Wallace — APPROVED** [review]
LGTM

File: src/storage/src/hummock/sstable/bloom.rs (hunk 3 above)
Comment (KeXiangWang) [reply src/storage/src/hummock/sstable/bloom.rs]: done

File: src/storage/src/hummock/sstable/bloom.rs (hunk 2 above)
Comment (KeXiangWang) [reply src/storage/src/hummock/sstable/bloom.rs]: done

**Comment by KeXiangWang** [comment]
I'm doing some performance test for this PR. Please do not merge it for now. Thanks!

**Pull request merged**